event_sidecar/src/event_stream_server/sse_server.rs

```diff
@@ -700,6 +760,44 @@ mod tests {
         should_filter_out(&finality_signature, &SIDECAR_FILTER[..]).await;
         should_not_filter_out(&shutdown, &SIDECAR_FILTER).await;
         should_not_filter_out(&sidecar_api_version, &SIDECAR_FILTER[..]).await;
+
+        // `EventFilter::Main` should only filter out `DeployAccepted`s and `FinalitySignature`s.
+        should_not_filter_out(&api_version, &MAIN_FILTER[..]).await;
+        should_not_filter_out(&block_added, &MAIN_FILTER[..]).await;
+        should_not_filter_out(&transaction_processed, &MAIN_FILTER[..]).await;
+        should_not_filter_out(&transaction_expired, &MAIN_FILTER[..]).await;
+        should_not_filter_out(&fault, &MAIN_FILTER[..]).await;
+        should_not_filter_out(&step, &MAIN_FILTER[..]).await;
+        should_not_filter_out(&shutdown, &MAIN_FILTER).await;
+
+        should_filter_out(&transaction_accepted, &MAIN_FILTER[..]).await;
+        should_filter_out(&finality_signature, &MAIN_FILTER[..]).await;
+
+        // `EventFilter::DeployAccepted` should filter out everything except `ApiVersion`s and
+        // `DeployAccepted`s.
+        should_not_filter_out(&api_version, &DEPLOYS_FILTER[..]).await;
+        should_not_filter_out(&transaction_accepted, &DEPLOYS_FILTER[..]).await;
+        should_not_filter_out(&shutdown, &DEPLOYS_FILTER[..]).await;
+
+        should_filter_out(&block_added, &DEPLOYS_FILTER[..]).await;
+        should_filter_out(&transaction_processed, &DEPLOYS_FILTER[..]).await;
+        should_filter_out(&transaction_expired, &DEPLOYS_FILTER[..]).await;
+        should_filter_out(&fault, &DEPLOYS_FILTER[..]).await;
+        should_filter_out(&finality_signature, &DEPLOYS_FILTER[..]).await;
+        should_filter_out(&step, &DEPLOYS_FILTER[..]).await;
+
+        // `EventFilter::Signatures` should filter out everything except `ApiVersion`s and
+        // `FinalitySignature`s.
```

Comment:
```suggestion
        // `EventFilter::Signatures` should filter out everything except `ApiVersion`s, 
        // `FinalitySignature`s and `Shutdown`s.
```

However, I'd consider removing these comments completely because they usually rot and the code is self explanatory.

Reply:
you're right